bindata/network/ovn-kubernetes/ovnkube-relay.yaml

```diff
@@ -0,0 +1,164 @@
+---
+kind: DaemonSet
+apiVersion: apps/v1
+metadata:
+  name: ovnkube-relay
+  namespace: openshift-ovn-kubernetes
+  annotations:
+    kubernetes.io/description: |
+      This daemonset launches the ovn-kubernetes OVN relay networking components.
+    release.openshift.io/version: "{{.ReleaseVersion}}"
+spec:
+  selector:
+    matchLabels:
+      app: ovnkube-relay
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 10%
+  template:
+    metadata:
+      annotations:
+        target.workload.openshift.io/management: '{"effect": "PreferredDuringScheduling"}'
+      labels:
+        app: ovnkube-relay
+        component: network
+        type: infra
+        openshift.io/component: network
+        kubernetes.io/os: "linux"
+    spec:
+      serviceAccountName: ovn-kubernetes-node
+      hostNetwork: true
+      hostPID: true
+      priorityClassName: "system-node-critical"
+      # volumes in all containers:
+      # (container) -> (host)
+      # /etc/openvswitch -> /etc/openvswitch - ovsdb system id
+      # /var/lib/openvswitch -> /var/lib/openvswitch/data - ovsdb data
+      # /run/openvswitch -> tmpfs - ovsdb sockets
+      # /env -> configmap env-overrides - debug overrides
+      containers:
+      # ovn-relay: programs the vswitch with flows from the sbdb
+      - name: ovn-relay
+        image: "{{.OvnImage}}"
+        command:
+        - /bin/bash
+        - -c
+        - |
+          set -e
+          if [[ -f "/env/${K8S_NODE}" ]]; then
+            set -o allexport
+            source "/env/${K8S_NODE}"
+            set +o allexport
+          fi  
+          
+          echo "$(date -Iseconds) - starting ovsdb-server as ovn-relay"
+          ovsdb-server -vconsole:emer -vsyslog:err -vfile:info \
+            --no-chdir \
+            -p /ovn-cert/tls.key -c /ovn-cert/tls.crt -C /ovn-ca/ca-bundle.crt \
+            --unixctl=/var/run/ovn/ovnsb_relay_db.ctl \
+            --user openvswitch:hugetlbfs \
```

Comment:
I don't think you need --user at all; looking at a live running sbdb (which is started by ovn-ctl) we see:

```
ovsdb-server -vconsole:info -vfile:off
--log-file=/var/log/ovn/ovsdb-server-sb.log
--remote=punix:/var/run/ovn/ovnsb_db.sock
--pidfile=/var/run/ovn/ovnsb_db.pid
--unixctl=/var/run/ovn/ovnsb_db.ctl
--disable-file-column-diff
--remote=db:OVN_Southbound,SB_Global,connections
--private-key=/ovn-cert/tls.key
--certificate=/ovn-cert/tls.crt
--ca-cert=/ovn-ca/ca-bundle.crt
--ssl-protocols=db:OVN_Southbound,SSL,ssl_protocols
--ssl-ciphers=db:OVN_Southbound,SSL,ssl_ciphers
/etc/ovn/ovnsb_db.db
```

That may be the cause of some of your permissions issues.